instrumentation/opentelemetry-instrumentation-flask/src/opentelemetry/instrumentation/flask/__init__.py

```diff
@@ -101,15 +103,17 @@ def _start_response(status, response_headers, *args, **kwargs):
                         "missing at _start_response(%s)",
                         status,
                     )
-
+                if otel_response_hook:
+                    otel_response_hook(span, status, response_headers) 
+            
             return start_response(status, response_headers, *args, **kwargs)
 
         return wsgi_app(wrapped_app_environ, _start_response)
 
     return _wrapped_app
 
 
-def _wrapped_before_request(name_callback):
+def _wrapped_before_request(name_callback, otel_request_hook=None):
```

Comment:
The arg should just be `request_hook`. No need to prefix otel_ here as we are already in otel context at this point.

Reply:
@owais 
Curious at how the hooks can replace the functionality of `name_callback` when a span's name cannot be modified after creation?